### DISCUSSION TITLE
Alternative to using `enabled` property of photo when using Metashape multispectral projects

### DISCUSSION THREAD

**Discussion by Ken-Kuroki** [UTokyo-FieldPhenomics-Lab/EasyIDP#77] [Q&A] [answered]
Hi!

When I tried EasyIDP with Metashape multispectral projects (captured with P4M), I found 

```python
ms = easyidp.Metashape(proj_path, chunk_id=0)
enabled_photos = [photo for photo in ms.photos if photo.enabled]
```

only keeps a small portion of images, and to be exact only blue band images like the following:

- DJI_0001.TIF - blue and enabled
- DJI_0002.TIF - green and not enabled
- DJI_0003.TIF - red and not enabled
- DJI_0004.TIF - rededge and not enabled
- DJI_0005.TIF - nir and not enabled
- DJI_0011.TIF - blue and enabled
- DJI_0012.TIF - green and not enabled
- DJI_0013.TIF - red and not enabled
- DJI_0014.TIF - rededge and not enabled
- DJI_0015.TIF - nir and not enabled
- ...

It seems like for all other bands `enabled` is `False`. I can't just ignore this because there are really unused images that need to be  excluded.

On Metashape these images are bandled into five-image sets represented by blue image, and this is probably why EasyIDP sees the rest of colors not enabled. I can implement a workaround to check their status as a set in my code, but I'd appreciate if you could tell me any smarter way to do it. Thanks!

**Comment by HowcanoeWang** [comment, marked as answer]
Currently EasyIDP is not designed for dealing with multispecral projects, could you share that project for testing?

Or just for this task you mentioned here, if you can ensure that all imageset are 5 tif files, you can do this way:

```python
enabled_photos = []
root_enabled = False:
for pid, photo in enumerate(ms.photos):
    # 0 % 5 = 0, 1 % 5 = 1, 
    # 5 % 5 = 0, 6 % 5 = 1, ...
    if pid % 5 == 0:   # the start photo
        if photo.enabled:
            root_enabled = True
        else:
            root_enabled = False
    if root_enabled:
        enabled_photos.append(photo)
```

It will pick up all enabled photos that you want

**Reply by Ken-Kuroki** [reply]
Thanks a lot! I eventually decided to overwrite `enabled` status of photos using the logic you suggested, because `idp.Metashape().back2raw()` uses this attribute to select the photos to do backprojection. It now seems to work.
